flexget/plugins/daemon/irc.py

```diff
@@ -375,15 +380,19 @@ def run_tasks(self):
             tasks_entry_map = {}
             for entry in self.entry_queue:
                 matched = False
-                for task, config in tasks_re.items():
-                    if isinstance(config, dict):
-                        config = [config]
-                    for c in config:
-                        if re.search(c['regexp'], entry.get(c['field'], ''), re.IGNORECASE):
-                            matched = True
-                            if not tasks_entry_map.get(task):
-                                tasks_entry_map[task] = []
-                            tasks_entry_map[task].append(entry)
+                for task_config in tasks_re:
+                    pattern_match = 0
+                    for pattern in task_config.get('patterns'):
```

Comment:
You've specified `one_or_more` in the schema, so this will fail if it's not a list.